src/contracts/core/AVSDirectory.sol

```diff
@@ -52,6 +55,110 @@ contract AVSDirectory is
+        require(
+            operatorSignature.expiry >= block.timestamp,
+            "AVSDirectory.registerOperatorToOperatorSet: operator signature expired"
+        );
+        require(
+            operatorSetRegistrations[msg.sender][operator][operatorSetID] != true,
+            "AVSDirectory.registerOperatorToOperatorSet: operator already registered to operator set"
+        );
+        require(
+            !operatorSaltIsSpent[operator][operatorSignature.salt],
+            "AVSDirectory.registerOperatorToAVS: salt already spent"
+        );
+        require(
+            delegation.isOperator(operator),
+            "AVSDirectory.registerOperatorToAVS: operator not registered to EigenLayer yet"
+        );
+        
+        // Calculate the digest hash
+        bytes32 operatorSetRegistrationDigestHash = calculateOperatorAVSRegistrationDigestHash({
+            operator: operator,
+            avs: msg.sender,
+            operatorSetID: operatorSetID,
+            salt: operatorSignature.salt,
+            expiry: operatorSignature.expiry
+        });
+
+        // Check that the signature is valid
+        EIP1271SignatureUtils.checkSignature_EIP1271(
+            operator,
+            operatorSetRegistrationDigestHash,
+            operatorSignature.signature
+        );
+
+        // Set as Operator Set AVS, preventing any further legacy registrations
+        if (!isOperatorSetAVS[msg.sender]) {
+            isOperatorSetAVS[msg.sender] = true;
+        }
+
+        // Update operator set registration
+        operatorSetRegistrations[msg.sender][operator][operatorSetID] = true;
+        operatorAVSOperatorSetCount[msg.sender][operator] += 1;
+
+        // Mark the salt as spent
+        operatorSaltIsSpent[operator][operatorSignature.salt] = true;
+
+        // Set the operator as registered if not already
+        if (avsOperatorStatus[msg.sender][operator] != OperatorAVSRegistrationStatus.REGISTERED) {
+            avsOperatorStatus[msg.sender][operator] = OperatorAVSRegistrationStatus.REGISTERED;
+            emit OperatorAVSRegistrationStatusUpdated(operator, msg.sender, OperatorAVSRegistrationStatus.REGISTERED);
+        }
+
+        emit OperatorAddedToOperatorSet(operator, msg.sender, operatorSetID);
+    }
+
+    /**
+     * @notice Called by AVSs or operators to remove an operator to from operator set
+     * 
+     * @param operator the address of the operator to be removed from the 
+     * operator set
+     * @param operatorSetID the ID of the operator set
+     * 
+     * @dev msg.sender is used as the AVS
+     * @dev operator must be registered for msg.sender AVS and the given 
+     * operator set
+     */
+    function deregisterOperatorFromOperatorSet(
+        address operator,
+        uint32 operatorSetID
+    ) external onlyWhenNotPaused(PAUSED_OPERATOR_REGISTER_DEREGISTER_TO_AVS) {
+        require(
+            operatorSetRegistrations[msg.sender][operator][operatorSetID] == true,
```

Comment:
u can get rid `== true`

Reply:
we are still not allowing the operator to call this?

Reply:
Correct @scotthconner . Operators still have to register via the AVS. We are considering adding functionality for operators to list the operator sets they want in this contract - "standby mode"